src/main/java/ke/co/proxyapi/jobnotificationparser/utils/AppUtils.java

```diff
@@ -35,4 +53,53 @@ public void sendToTelegram(String urlStr)
 			template.asyncSendBody("direct:telegram", advert);
 		}
 	}
+
+	private boolean hasSpamWords(String url) {
+		String description = getMetaDescription(url);
+		if (description == null) {
+			return false;
+		}
+		return isSpamDescription(description, "spam_words.txt");
+	}
+	public static String getMetaDescription(String url) {
+		try {
+			Document doc = Jsoup.connect(url).get();
+			Elements metaTags = doc.getElementsByTag("meta");
+			for (Element metaTag : metaTags) {
+				String tagName = metaTag.attr("name");
+				if (tagName.equalsIgnoreCase("description")) {
+					return metaTag.attr("content");
+				}
+			}
+		} catch (IOException ignored) {
+		}
+		return null;
+	}
+	public static boolean isSpamDescription(String description, String spamWordsFilePath) {
+		String[] spamWords = readSpamWordsFromFile(spamWordsFilePath);
+
+		for (String spamWord : spamWords) {
+			Pattern pattern = Pattern.compile(spamWord, Pattern.CASE_INSENSITIVE| Pattern.LITERAL | Pattern.MULTILINE | Pattern.DOTALL);
+			Matcher matcher = pattern.matcher(description);
+			if (matcher.find()) {
+				return true;
+			}
+		}
+		return false;
+	}
+	private static String[] readSpamWordsFromFile(String spamWordsFilePath) {
+		//read spam words from resources/blacklist.json
+		try
+		{
+			ClassPathResource res = new ClassPathResource(spamWordsFilePath);
```

Comment:
Since the spamwords file is only used once here, just remove it from the rest of the method params and insert it here directly. Better yet, learn how to inject it as a variable using [Autowiring](https://www.baeldung.com/spring-value-annotation) as it can be changed anytime